dbcon/mysql/ha_mcs_ddl.cpp

```diff
@@ -2583,7 +2588,7 @@ int ha_mcs_impl_delete_table_(const char* db, const char* name, cal_connection_i
     stmt += ";";
     int rc = ProcessDDLStatement(stmt, schema, tbl, tid2sid(thd->thread_id), emsg);
 
-    if (rc != 0)
+    if (rc != 0 && rc != ER_NO_SUCH_TABLE_IN_ENGINE)
```

Comment:
Do we not need to push_warning for other errors as well?

Reply:
We are pushing warnings for all errors, except when rc=ER_NO_SUCH_TABLE_IN_ENGINE. If I include rc=ER_NO_SUCH_TABLE_IN_ENGINE in the push_warning, I get the following:
```sql
MariaDB [test]> drop table if exists tttttt;
Query OK, 0 rows affected, 2 warnings (0.085 sec)

MariaDB [test]> show warnings;
+---------+------+--------------------------------------------------+
| Level   | Code | Message                                          |
+---------+------+--------------------------------------------------+
| Warning | 9999 | CAL0001: Table does not exist in ColumnStore.    |
| Note    | 1051 | Unknown table 'test.tttttt'                      |
+---------+------+--------------------------------------------------+
2 rows in set (0.000 sec)
```

The expected warning here should be:
```sql
MariaDB [test]> show warnings;
+-------+------+---------------------------+
| Level | Code | Message                   |
+-------+------+---------------------------+
| Note  | 1051 | Unknown table 'test.tttt' |
+-------+------+---------------------------+
```

This is why rc=ER_NO_SUCH_TABLE_IN_ENGINE is excluded.

Reply:
ah I see.  I misread this the first time I looked at it.  I agree, this should warn on everything but no-such-table, which is handled a different way.